telegram/tests/test_telegram.py

```diff
@@ -0,0 +1,394 @@
+
+    # Telegram chat types
+    PRIVATE = 'private'
+    CHANNEL = 'channel'
+    GROUP = 'group'
+
+    # Some default Telegram objects
+    default_user = {
+        'id': 'default_user_id',
+        'username': '@default_user',
+    }
+    bad_telegram_response = {
+        'ok': False,
+        'description': 'Bad request',
+    }
+
+    @inlineCallbacks
+    def setUp(self):
+        self.helper = self.add_helper(
+            HttpRpcTransportHelper(TelegramTransport)
+        )
+        self.request_queue = DeferredQueue()
+        self.pending_requests = []
+        self.addCleanup(self.finish_requests)
+        self.mock_server = FakeHttpServer(self.handle_inbound_request)
+
+        self.transport = yield self.get_transport()
+
+    @inlineCallbacks
+    def get_transport(self, **config):
+        defaults = {
+            'bot_username': self.bot_username,
+            'bot_token': self.TOKEN,
+            'web_path': 'foo',
+            'web_port': 0,
+            'inbound_url': 'www.example.com',
+            'outbound_url': self.API_URL
+        }
+        defaults.update(config)
+        transport = yield self.helper.get_transport(defaults)
+        transport.agent_factory = self.mock_server.get_agent
+        returnValue(transport)
+
+    def handle_inbound_request(self, req):
+        self.request_queue.put(req)
+        return NOT_DONE_YET
+
+    @inlineCallbacks
+    def get_next_request(self):
+        req = yield self.request_queue.get()
+        self.pending_requests.append(req)
+        returnValue(req)
+
+    @inlineCallbacks
+    def finish_requests(self):
+        for req in self.pending_requests:
+            if not req.finished:
+                yield req.finish()
+
+    def test_get_outbound_url(self):
+        """
+        Our helper method for building outbound URLs should build URLs using
+        the Telegram API URL, our bot token, and the applicable method (path)
+        """
+        test_url = self.transport.get_outbound_url('myPath')
+        self.assertEqual(test_url, '%s%s/%s' %
+                         (self.API_URL, self.TOKEN, 'myPath'))
+
+    @inlineCallbacks
+    def test_setup_webhook_no_errors(self):
+        """
+        We should log successful webhook setup and our request should be
+        in the proper format
+        """
+        d = self.transport.setup_webhook()
+        expected_url = '%s%s/%s' % (self.API_URL.rstrip('/'), self.TOKEN,
+                                    'setWebhook')
+
+        req = yield self.get_next_request()
+        self.assertEqual(req.method, 'POST')
+        self.assertEqual(req.path, expected_url)
+
+        content = json.load(req.content)
+        self.assertEqual(content['url'], 'www.example.com')
+
+        req.write(json.dumps({'ok': True}))
+        with LogCatcher(message='Webhook') as lc:
+            req.finish()
+            yield d
+            [log] = lc.messages()
+            self.assertEqual(log, 'Webhook set up on www.example.com')
```

Comment:
You can also do this as `self.assertEqual(lc.messages(), ['Webhook set up on www.example.com'])` which has the very slight advantage of showing the extra log messages if there are any.